lib/analyzer/applications/java.ts

```diff
@@ -0,0 +1,58 @@
+import * as path from "path";
+import { JarFingerprint } from "../types";
+import { AppDepsScanResultWithoutTarget, FilePathToContent } from "./types";
+
+function mapContentToJarFingerprints(input: {
+  [fileName: string]: string;
+}): {
+  [path: string]: JarFingerprint[];
+} {
+  const jarFingerprints: JarFingerprint[] = Object.entries(input).map(
+    ([filePath, digest]) => {
+      return {
+        location: filePath,
+        digest,
+      };
+    },
+  );
+
+  const resultAggregatedByPath = jarFingerprints.reduce(
+    (accumulator, current) => {
+      const location = path.dirname(current.location);
+      accumulator[location] = accumulator[location] || [];
+      accumulator[location].push(current);
+      return accumulator;
+    },
+    {},
+  );
+
+  return resultAggregatedByPath;
+}
+
+export async function jarFilesToScannedProjects(
+  filePathToContent: FilePathToContent,
+): Promise<AppDepsScanResultWithoutTarget[]> {
+  const mappedResult = mapContentToJarFingerprints(filePathToContent);
+  const scanResults: AppDepsScanResultWithoutTarget[] = [];
+
+  for (const key in mappedResult) {
+    if (!Object.prototype.hasOwnProperty.call(mappedResult, key)) {
```

Comment:
It seems odd that we have to call this at all. Maybe the `for loop` is giving us the wrong data? Maybe instead of `for ... in ...` we need `for ... of ...`? Or we could use something like `Object.entries()`?

Reply:
yes, changed 👍